lib/workers/bin/run_single_worker.rb

```diff
@@ -52,7 +56,12 @@
 worker_class = worker_class.constantize
 worker_class.before_fork
 unless options[:dry_run]
-  worker = worker_class.create_worker_record
+  worker = if options[:guid]
+             worker_class.find_by(:guid => options[:guid])
```

Comment:
Since I wrote this, I can be critical of it ;)

Maybe we should do a `find_by!`, or something similar so that we error out if we can't find the worker record.